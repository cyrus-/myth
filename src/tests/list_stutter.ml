let s = "
type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let list_stutter : list -> list |>
  { [] => []
  | [0] => [0;0]
  | [1;0] => [1;1;0;0]
  } = ?"
