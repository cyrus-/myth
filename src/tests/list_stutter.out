(* Generated Thu Nov 13 00:35:52 2014 *)
let list_stutter : list -> list =
  let rec f1 (l1:list) : list =
    match l1 with
      | Nil -> Nil
      | Cons (n1, l2) -> Cons (n1, Cons (n1, f1 l2))
  in
    f1
;;
